### ISSUE TITLE
too many positional arguments

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1422]
(working around #1356)
looks like `/* */` should be accepted as comment characters???
```
$ ARCH=s390 CROSS_COMPILE=s390x-linux-gnu- make -j72 CC=clang LLVM_IAS=1 -j72 lib/raid6/s390vx8.o
<instantiation>:3:21: error: too many positional arguments
        GR_NUM  b2, %r1     /* Base register */
                            ^
lib/raid6/s390vx8.c:63:16: note: while in macro instantiation
        asm volatile ("VLM %2,%3,0,%1"
                      ^
<inline asm>:1:2: note: instantiated into assembly here
        VLM 16,23,0,%r1
        ^
```

**Comment by uweigand** [comment]
From what I can see, the clang inline assembler normally accepts `/* ... */` comments.   It seems this particular error is related to the fact that the comment occurs in a macro invocation (and, not sure if that matters, that is nested within another macro definition).

That doesn't appear platform-specific, but seems a deficiency in LLVM's general support for assembler macros.

**Comment by nickdesaulniers** [comment]
Shouldn't the C preprocessor be dropping these comments? Perhaps this .h file is being included in a .s file rather than a .S?

Oh, boy, lib/raid6/s390vx.uc is including arch/s390/include/asm/vx-insn.h via `.include` assembler directive, from top level inline asm in C...

**Comment by ihalip** [comment]
What about just removing the comments?
```diff
diff --git a/arch/s390/include/asm/vx-insn.h b/arch/s390/include/asm/vx-insn.h
index 0c05a673811c..2b8f1f82ef1a 100644
--- a/arch/s390/include/asm/vx-insn.h
+++ b/arch/s390/include/asm/vx-insn.h
@@ -366,7 +366,7 @@
 .macro VLM     vfrom, vto, disp, base, hint=3
        VX_NUM  v1, \vfrom
        VX_NUM  v3, \vto
-       GR_NUM  b2, \base           /* Base register */
+       GR_NUM  b2, \base
        .word   0xE700 | ((v1&15) << 4) | (v3&15)
        .word   (b2 << 12) | (\disp)
        MRXBOPC \hint, 0x36, v1, v3
@@ -376,7 +376,7 @@
 .macro VSTM    vfrom, vto, disp, base, hint=3
        VX_NUM  v1, \vfrom
        VX_NUM  v3, \vto
-       GR_NUM  b2, \base           /* Base register */
+       GR_NUM  b2, \base
        .word   0xE700 | ((v1&15) << 4) | (v3&15)
        .word   (b2 << 12) | (\disp)
        MRXBOPC \hint, 0x3E, v1, v3
```

**Comment by nathanchance** [comment]
Fix queued: https://git.kernel.org/s390/c/964bc5dbe602a62b7bbd67624c4b8f7a4ea692b2

**Comment by nickdesaulniers** [comment]
merged in 5.19-rc1 964bc5dbe602a62b7bbd67624c4b8f7a4ea692b2
